### ISSUE TITLE
Query

### ISSUE THREAD

**Issue by masfoobar** [laytan/odin-mysql#3]
Hi,

This issue covers: -
   Request: Improve on Readme.md
   Issue: Potential bug or (likely) user misuse.

I am new to odin and using this library to help learn (and because I want to write a home project which will use MySQL and would love it to be using odin) :-)

Is it possible for readme.md to contain a simple example of usage - such as queryinh a fictitious database?

When I try something like below, nothing gets printed when I run it.

```package main

import "core:fmt"
import mysql "./odin-mysql"

main :: proc() {
    fmt.println("Helloooo")

    info := string(mysql.get_client_info())
    fmt.println(info);
}
```

Expected outcome:
> Hellooo
> {0}.{0}.{0}             # the version number like 9.1.0

Actual outcome print nothing

I have tried variations (without string cast, etc) and the same result. When I comment out the last 2 lines in main, "Hellooo" is printed.

Again, I am learning odin and likely a user error than anything. However, I think it would still be good to have an example as explained earlier.

I tried writing other things to rule out, like mysql.init, mysql.connect, mysql.query (to create a new database) but nothing. the user I am logging in with have full rights.

Steps taken to create the above:-
- Create new folder (dbtest)
- In dbtest
-- Create main.odin
-- Paste said code above into main.odin
-- git clone odin-mysql   (creates odin-mysql folder inside dbtest)
-- add mysqlclient.lib

- odin build .
- ./dbtest

Thank you.

**Comment by laytan** [comment]
Hey, yeah the repo needs some examples I agree and will try to add some soon.

In the meantime, this is wrapping the official API of mysql client in C. This means that everything in the docs here apply: https://dev.mysql.com/doc/c-api/8.0/en

The differences are that the function names in c all have `mysql_` prefixed, for this package that is `mysql.` instead (using the packaging feature in Odin). And the mysql functions with `real_` in them have that removed.

This is a pretty nice overview in particular https://dev.mysql.com/doc/c-api/8.0/en/c-api-basic-interface-usage.html

General example, this also uses some special wrappers around prepared statements:
```odin
get_user_by_email :: proc(email: string) -> (id: i32, name: string) {
	db := mysql.init() // I believe your code above doesn't work because you need to call this before using anything out of the package.
	assert(db != nil)
	defer mysql.library_end()

	fmt.println(mysql.get_client_info()) // Should now give you something.

	if mysql.connect(db, "127.0.0.1", "root", "root", "mydatabase") == nil {
		panic(string(mysql.error(db)))
	}
	defer mysql.close(db)

	stmt := mysql.stmt_init(db)
	if stmt == nil {
		panic(string(mysql.error(db)))
	}
	defer mysql.stmt_close(stmt)

	query: cstring : "SELECT id, name FROM users where email = ?"
	if mysql.stmt_prepare(stmt, query, u64(len(query))) != 0 {
		panic(string(mysql.stmt_error(stmt)))
	}

	binds: [1]mysql.Bind
	b, l := mysql.bindp_var_char("example@email.com")
	binds[0] = b
	defer free(l)

	if mysql.stmt_bind_param(stmt, raw_data(binds[:])) {
		panic(string(mysql.stmt_error(stmt)))
	}

	if mysql.stmt_execute(stmt) != 0 {
		panic(string(mysql.stmt_error(stmt)))
	}

	rbinds: [2]mysql.Bind

	rbinds[0] = mysql.bindr_int(&id)

	name_len: c.ulong // need import "core:c"
	rbinds[1] = mysql.bindr_var_char(nil, &name_len)

	if mysql.stmt_bind_result(stmt, raw_data(rbinds[:]))  {
		panic(string(mysql.stmt_error(stmt)))
	}

	if mysql.stmt_store_result(stmt) != 0 {
		panic(string(mysql.stmt_error(stmt)))
	}

	status := mysql.stmt_fetch(stmt)
	switch status {
	case mysql.NO_DATA: panic("not found") // probably return some kind of boolean
	case mysql.DATA_TRUNCATED, 0: // ok
	case: panic(string(mysql.stmt_error(stmt)))
	}

	name_buf := make([]byte, int(name_len))
	rbinds[1].buffer = raw_data(name_buf)
	rbinds[1].buffer_length = name_len
	if mysql.stmt_fetch_column(stmt, raw_data(rbinds[:]), 1, 0) != 0 {
		panic(string(mysql.stmt_error(stmt)))
	}
	
	name = string(name_buf)
	return
}
```

I have not verified the above to work as I just put this together, but you should get pretty far.

The API is pretty hard to use in general because they try to offer as much control as possible and expect people to create wrapper functions themselves (ORMs for example).

**Comment by masfoobar** [comment]
Thank you for the reply.

Yes I managed to piece together the C equivalent to a few of the functions needed to get it working. I did originally include mysql.init --- I guess I removed it while trying variations. 

I will continue playing around tomorrow. Your example will be very helpful, so thanks.

I have played with the C api a few years ago (in C) - so if I can get it working in Odin I will likely experiment building wrappers around it.
